quil-rs/src/expression/simplification.rs

```diff
@@ -476,6 +476,8 @@ static RULES: Lazy<Vec<Rewrite>> = Lazy::new(|| {
         // + & *
         rw!("distribute"    ; "(* ?a (+ ?b ?c))"        => "(+ (* ?a ?b) (* ?a ?c))"),
         rw!("factor"        ; "(+ (* ?a ?b) (* ?a ?c))" => "(* ?a (+ ?b ?c))"),
+        rw!("commute-add"   ; "(+ (+ ?a ?b) ?c)"        => "(+ ?a (+ ?b ?c))"),
```

Comment:
Nit: this is associativity, not commutativity. Commutativity would be `(+ ?a ?b)` => `(+ ?b ?a)`.

Reply:
⬆️

Reply:
Same below

Reply:
doh. fixed